doc/whats_new.rst

```diff
@@ -212,6 +212,9 @@ Enhancements
      (`#6846 <https://github.com/scikit-learn/scikit-learn/pull/6846>`_)
      By `Sebastian Säger`_ and `YenChen Lin`_.
 
+   - Isotonic regression (:mod:`isotonic`) now uses a better algorithm to avoid
+     `O(n^2)` behavior in pathological cases, and is also generally faster.
```

Comment:
Please add a link to this PR. Usually we also mention the author name, unless you would rather not.